pkgs/servers/home-assistant/default.nix

```diff
@@ -179,10 +179,13 @@ in with py.pkgs; buildPythonApplication rec {
 
   postPatch = ''
     substituteInPlace setup.py \
+      --replace "async_timeout==3.0.1" "async_timeout" \
+      --replace "awesomeversion==21.10.1" "awesomeversion" \
+      --replace "aiohttp==3.7.4.post0" "aiohttp" \
       --replace "bcrypt==3.1.7" "bcrypt" \
       --replace "pip>=8.0.3,<20.3" "pip" \
       --replace "pyyaml==6.0" "pyyaml" \
-      --replace "yarl==1.6.3" "yarl==1.7.0"
+      --replace "yarl==1.6.3" "yarl"
```

Comment:
This is just me thinking loudly here, but maybe we could have a function to relax Python dependencies (or even something that is part of `buildPython{Module,Application}`) since this is a pretty common pattern in nixpkgs :thinking: ?

Something like:

```
relaxSetuppyDependencies = [
  "async_timeout"
  "awesomeversion"
  ...
];
```

Reply:
Sometimes they're in setup.py, sometimes in setup.cfg, sometimes sourced from requirements.txt or worse from python code somewhere in the repository, in newer projects in pyproject.toml and other files. It's a good idea, but gets hairy fast.

Reply:
Yeah, but we could have something generic since they pretty much use the same syntax to declare its values. Maybe as a function available to Python derivations like we already have the `substituteInPlace`, so we could have maybe `relaxPythonDependencies setup.cfg ...` instead.

Reply:
And of course, you can still use `substituteInPlace` or even `sed` if necessary for those more strange cases.